packages/terra-button/package.json

```diff
@@ -42,5 +42,8 @@
     "keycode-js": "^2.0.1",
     "prop-types": "^15.5.8",
     "terra-icon": "^3.27.0"
+  },
+  "devDependencies": {
+    "terra-abstract-modal": "^3.21.0"
```

Comment:
The files in terra-dev-site directory are released with the component. This will have to be a hard dependency if we want to keep the example. We'll either need to remove the example or move this to be a hard dependency.

Reply:
changed here b5ea39252dacda797d3d046bf3c4a1e6183b80e3

Reply:
We should get a few more opinions on this. Adding a transitive dependency just for testing is usually something we should try to avoid when possible.

cc: @tbiethman @mjhenkes

Reply:
Terra-button should not have a dependency on terra-abstract-modal.

Reply:
we could implement this example using html-dialog. to avoid the hard dependency on terra-abstract-modal. similar to what we have done for [Hookshot](https://engineering.cerner.com/terra-ui/components/terra-hookshot/hookshot/hookshot) examples before. by removing the hard dependency on terra-form-select.

Reply:
@StephenEsser @mjhenkes @ShettyAkarsh  updated here 716688cc6b76a7f8f6ba276ed52158a5779e87f5